ethereum_eip1559_transaction.go

```diff
@@ -0,0 +1,167 @@
+		item.childItems[3].itemValue,
+		item.childItems[4].itemValue,
+		item.childItems[5].itemValue,
+		item.childItems[6].itemValue,
+		item.childItems[7].itemValue,
+		item.childItems[9].itemValue,
+		item.childItems[10].itemValue,
+		item.childItems[11].itemValue,
+		accessListValues,
+	), nil
+}
+
+// ToBytes encodes the EthereumEIP1559Transaction into RLP format.
+func (txn *EthereumEIP1559Transaction) ToBytes() ([]byte, error) {
+	item := NewRLPItem(LIST_TYPE)
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.ChainId))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.Nonce))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.MaxPriorityGas))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.MaxGas))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.GasLimit))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.To))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.Value))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.CallData))
+	accessListItem := NewRLPItem(LIST_TYPE)
+	for _, itemBytes := range txn.AccessList {
+		accessListItem.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(itemBytes))
+	}
+	item.PushBack(accessListItem)
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.RecoveryId))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.R))
+	item.PushBack(NewRLPItem(VALUE_TYPE).EIP1559TransactionAssignBytes(txn.S))
+
+	transactionBytes, err := item.Write()
+	if err != nil {
+		return nil, err
+	}
+	// Append 02 byte as it is the standard for EIP1559
+	combinedBytes := append([]byte{0x02}, transactionBytes...)
+
+	return combinedBytes, nil
+}
+
+// String returns a string representation of the EthereumEIP1559Transaction.
+func (txn *EthereumEIP1559Transaction) String() string {
+	// Encode each element in the AccessList slice individually
+	var encodedAccessList []string
+	for _, entry := range txn.AccessList {
+		encodedAccessList = append(encodedAccessList, hex.EncodeToString(entry))
+	}
+
+	accessListStr := "[" + strings.Join(encodedAccessList, ", ") + "]"
+
+	return fmt.Sprintf("ChainId: %s\nNonce: %s\nMaxPriorityGas: %s\nMaxGas: %s\nGasLimit: %s\nTo: %s\nValue: %s\nCallData: %s\nAccessList: %s\nRecoveryId: %s\nR: %s\nS: %s",
+		hex.EncodeToString(txn.ChainId),
+		hex.EncodeToString(txn.Nonce),
+		hex.EncodeToString(txn.MaxPriorityGas),
+		hex.EncodeToString(txn.MaxGas),
+		hex.EncodeToString(txn.GasLimit),
+		hex.EncodeToString(txn.To),
+		hex.EncodeToString(txn.Value),
+		hex.EncodeToString(txn.CallData),
+		accessListStr,
+		hex.EncodeToString(txn.RecoveryId),
+		hex.EncodeToString(txn.R),
+		hex.EncodeToString(txn.S),
+	)
+}
+
+// Helper function to create an RLP item for value assignment
+func (item *RLPItem) EIP1559TransactionAssignBytes(value []byte) *RLPItem {
```

Comment:
This is the same as `LegacyTransactionAssignBytes` right?
If so - I think it can be moved to `rlp.go` (since this func belongs to `RLPItem`) and rename it to something like `func (item *RLPItem) AssignBytes(value []byte) *RLPItem {}`

Reply:
Now i see that we already have `AssignBytes` func. Maybe you can make it return an `*RLPItem` and it would do the job, i think.